Adding upgrade notes for 0.21 release (#1116)
luckyframework/lucky: UPGRADE_NOTES.md

## Upgrading from 0.20 to 0.21

- Upgrade Lucky CLI (homebrew)

```
brew update
brew upgrade lucky
```

- Upgrade Lucky CLI (Linux)

> Remove the existing Lucky binary and follow the Linux
> instructions in this section
> https://luckyframework.org/guides/getting-started/installing#on-linux

- Update versions in `shard.yml`
  - Crystal should be `0.34.0`
  - Lucky should be `~> 0.21.0`
  - Authentic should be `~> 0.5.4`
  - LuckyFlow should be `~> 0.6.3`

- Run `shards update`

### General updates

- Rename: `config/logger.cr` to `config/log.cr`

> - Update: `config/log.cr` to use the new `Log`. [See implementation](https://github.com/luckyframework/lucky_cli/blob/master/src/web_app_skeleton/config/log.cr)

Reviewer: There is also an update to `config/database.cr` that remove the comments and code about the query log config: 

https://github.com/luckyframework/lucky_cli/commit/1f2015eaea0883ec7b5863cc7527be344faa636a#diff-15fb0661e7cd546c3789df4bdbf83cbcL26

So they should delete the comment and if they want to log all queries they need to instead change: https://github.com/luckyframework/lucky_cli/commit/1f2015eaea0883ec7b5863cc7527be344faa636a#diff-3c1593e181f651f222dded01d5d1ab5eR35

- Update: `Procfile.dev` and update the `system_check` to `script/system_check && sleep 100000`.

> - Update: all `Lucky.logger.whatever_level()` instances to use the new logging `Lucky::Log.dexter.whatever_level { }`

Reviewer: ```suggestion
- Update: all `Lucky.logger.{level}("message")` calls to use the new Crystal Log `Log.{level} { "message" }`
- Update: all `Lucky.logger.{level}(key: "value")` calls to use `Log.dexter{level} { {key: "value"} }`
```

> ### Updating the logging

Reviewer: ```suggestion
### Updating `Lucky.logger`
```

> Before this version, you would do log like this:

Reviewer: ```suggestion
Before this version, you would log data like this:
```

```crystal
Lucky.logger.debug("Logging some message")
Lucky.logger.info({path: @context.request.path})
```

Now, you would write this like:

```crystal
# Use the Crystal std-lib log for simple String messages
Log.debug { "Logging some message" }

> # Use the Dexter extention for more complex logging

Reviewer: ```suggestion
# Use the Dexter extension for logging key/value data
```
